multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -47,16 +74,338 @@ type ResourceExportReconciler struct {
+	var resExport mcsv1alpha1.ResourceExport
+	if err := r.Client.Get(ctx, req.NamespacedName, &resExport); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// We are using Finalizers to implement asynchronous pre-delete hooks.
+	// When ResourceExport is deleted, it will have non-zero DeletionTimestamp
+	// but controller can still get the deleted ResourceExport object, so we can
+	// clean up any external resources like ResourceImport.
+	// More details about using Finalizers, please refer to https://book.kubebuilder.io/reference/using-finalizers.html.
+	if !resExport.DeletionTimestamp.IsZero() {
+		if common.StringExistsInSlice(resExport.Finalizers, common.ResourceExportFinalizer) {
+			err := r.handleDeleteEvent(ctx, &resExport)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			resExport.SetFinalizers(common.RemoveStringFromSlice(resExport.Finalizers, common.ResourceExportFinalizer))
+			if err := r.Client.Update(ctx, &resExport, &client.UpdateOptions{}); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, nil
+	}
+
+	createResImport, existResImport, err := r.getExistingResImport(ctx, resExport)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	var changed bool
+	resImport := &mcsv1alpha1.ResourceImport{}
+	switch resExport.Spec.Kind {
+	case common.ServiceKind:
+		resImport, changed, err = r.refreshServiceResourceImport(&resExport, existResImport, createResImport)
+	case common.EndpointsKind:
+		resImport, changed, err = r.refreshEndpointsResourceImport(&resExport, existResImport, createResImport)
+	}
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	resImportName := types.NamespacedName{
+		Name:      getResourceImportName(&resExport),
+		Namespace: req.Namespace,
+	}
+
+	if createResImport {
+		if err = r.Client.Create(ctx, resImport, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "failed to create ResourceImport", "resourceimport", resImportName.String())
+			return ctrl.Result{}, err
+		}
+		klog.InfoS("create ResourceImport successfully", "resourceimport", resImportName.String())
+	} else if changed {
+		klog.InfoS("update ResourceImport for ResoureExport", "resourceimport", resImportName.String(), "resourceexport", req.NamespacedName)
+		if err = r.handleUpdateEvent(ctx, resImport, &resExport); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *ResourceExportReconciler) handleUpdateEvent(ctx context.Context,
+	resImport *mcsv1alpha1.ResourceImport, resExport *mcsv1alpha1.ResourceExport) error {
+	resImpName := types.NamespacedName{
+		Name:      getResourceImportName(resExport),
+		Namespace: resExport.Namespace,
+	}
+	var err error
+	if err = r.Client.Update(ctx, resImport, &client.UpdateOptions{}); err != nil {
+		klog.ErrorS(err, "failed to update ResourceImport", "resourceimport", resImpName.String())
```

Comment:
FYI. - in general, we do not log the error if it is returned and logged in the caller, unless we want to log with necessary context.
But you might have reasons to add these logs.

Reply:
yes, the main reason is Reconcile will print the log but hard to quickly locate where the err is from because of kubebuilder controller wrapper.